src/doc/trpl/documentation.md

```diff
@@ -1,224 +1,222 @@
 ```
-~~~
+/// # Examples
+///
+/// Simple `&str` patterns:
+///
+/// ```
+/// let v: Vec<&str> = "Mary had a little lamb".split(' ').collect();
+/// assert_eq!(v, vec!["Mary", "had", "a", "little", "lamb"]);
+/// ```
+///
+/// More complex patterns with a lambda:
+///
+/// ```
+/// let v: Vec<&str> = "abc1def2ghi".split(|c: char| c.is_numeric()).collect();
+/// assert_eq!(v, vec!["abc", "def", "ghi"]);
+/// ```
+# fn foo() {}
+```
 
-Lastly, you can specify that a code block be compiled as if `--test`
-were passed to the compiler using the `test_harness` directive.
+Let's discuss the details of these code blocks.
 
-~~~md
-```test_harness
-#[test]
-fn foo() {
-    panic!("oops! (will run & register as a failed test)")
-}
-```
-~~~
+#### Code block annotations
 
-Rustdoc also supplies some extra sugar for helping with some tedious
-documentation examples. If a line is prefixed with `# `, then the line
-will not show up in the HTML documentation, but it will be used when
-testing the code block (NB. the space after the `#` is required, so
-that one can still write things like `#[derive(Eq)]`).
+To write some Rust code in a comment, use the triple graves:
 
-~~~md
 ```
-# /!\ The three following lines are comments, which are usually stripped off by
-# the doc-generating tool.  In order to display them anyway in this particular
-# case, the character following the leading '#' is not a usual space like in
-# these first five lines but a non breakable one.
-# // showing 'fib' in this documentation would just be tedious and detracts from
-# // what's actually being documented.
-# fn fib(n: int) { n + 2 }
+/// ```
+/// println!("Hello, world");
+/// ```
+# fn foo() {}
+```
+
+If you want something that's not Rust code, you can add an annotation:
 
-spawn(move || { fib(200); })
 ```
-~~~
+/// ```c
+/// printf("Hello, world\n");
+/// ```
+# fn foo() {}
+```
+
+This will highlight according to whatever language you're showing off.
+If you're just showing plain text, choose `text`.
+
+It's important to choose the correct annotation here, because Rustdoc uses it
+in an interesting way: It can be used to actually test your examples, so that
+they don't get out of date. If you have some C code but Rustdoc thinks it's
+Rust because you left off the annotation, Rustdoc will complain when trying to
+generate the documentation.
 
-The documentation online would look like `spawn(move || { fib(200); })`, but when
-testing this code, the `fib` function will be included (so it can compile).
+Let's talk about that in more detail:
 
-Rustdoc will automatically add a `main()` wrapper around your code, and in the right
-place. For example:
+## Documentation as tests
+
+Let's discuss our sample example documentation:
+
+```
+/// ```
+/// println!("Hello, world");
+/// ```
+# fn foo() {}
+```
+
+You'll notice that you don't need a `fn main()` or anything here. Rustdoc will
+automatically add a main() wrapper around your code, and in the right place.
+For example:
```

Comment:
Maybe you could reuse the `println!` example given just above rather than introducing a new one?